AEPIdentity/Tests/IdentityFunctionalTests.swift

```diff
@@ -51,6 +73,61 @@ class IdentityFunctionalTests: XCTestCase {
     }
 
+    /// Tests that a sync event dispatches a shared state update
+    func testSyncIdentifiersForceSyncEventInstallScenario() {
+        // setup
+        setDefaultSuccessNetworkResponse()
+        let configSharedState = [IdentityConstants.Configuration.GLOBAL_CONFIG_PRIVACY: PrivacyStatus.optedIn.rawValue, IdentityConstants.Configuration.EXPERIENCE_CLOUD_ORGID: "test-org-id"]
+        let data = [IdentityConstants.EventDataKeys.IS_SYNC_EVENT: true]
+        let event = Event(name: "Sync Event", type: EventType.identity, source: EventSource.requestIdentity, data: data)
+        mockRuntime.simulateSharedState(extensionName: "com.adobe.module.configuration", event: event, data: (configSharedState, .set))
+        waitForProcessing()
+
+        // test
+        XCTAssertTrue(identity.readyForEvent(event)) //// since first event will be processed by readyForEvent which calls forceSync
+        waitForProcessing()
+
+        // verify
+        let sharedState = mockRuntime.createdSharedStates.last!
+        XCTAssertNotNil(sharedState?[IdentityConstants.EventDataKeys.VISITOR_ID_ECID])
+        XCTAssertTrue(mockNetworkService.connectAsyncCalled)
+        XCTAssertEqual(1, mockNetworkService.calledNetworkRequests.count)
+    }
+
+    /// Tests that a sync event dispatches a shared state update event and a sync network request on subsequent launch when forceSync flag is set
+    func testSyncIdentifiersForceSyncEventSubsequentLaunchScenario() {
+        // setup
+        setDefaultSuccessNetworkResponse()
+        let configSharedState = [IdentityConstants.Configuration.GLOBAL_CONFIG_PRIVACY: PrivacyStatus.optedIn.rawValue, IdentityConstants.Configuration.EXPERIENCE_CLOUD_ORGID: "test-org-id"]
+        let data = [IdentityConstants.EventDataKeys.IS_SYNC_EVENT: true]
+        let event = Event(name: "Sync Event", type: EventType.identity, source: EventSource.requestIdentity, data: data)
+        mockRuntime.simulateSharedState(extensionName: "com.adobe.module.configuration", event: event, data: (configSharedState, .set))
+
+        // test
+        XCTAssertTrue(identity.readyForEvent(event)) // since first event will be processed by readyForEvent which calls forceSync
+        waitForProcessing()
+
+        // verify
+        let sharedStateForInstall = mockRuntime.createdSharedStates.last!
+        XCTAssertNotNil(sharedStateForInstall?[IdentityConstants.EventDataKeys.VISITOR_ID_ECID])
+        XCTAssertTrue(mockNetworkService.connectAsyncCalled)
+        XCTAssertEqual(1, mockNetworkService.calledNetworkRequests.count)
+
+        let forceSyncdata = [IdentityConstants.EventDataKeys.IS_SYNC_EVENT: true, IdentityConstants.EventDataKeys.FORCE_SYNC: true]
+        let forceSyncEvent = Event(name: "Sync Event", type: EventType.identity, source: EventSource.requestIdentity, data: forceSyncdata)
+
+        // test
+        mockRuntime.simulateComingEvent(event: forceSyncEvent)
```

Comment:
This scenario cannot be reproduced in a real setup since we don't have the FORCE_SYNC key inside event data. How about we transform this in integration test and simulate the SDK restart mid test:  extensions registration + config, check identity sends force sync, reset event hub and re-init extensions, check Identity sends a force sync
This might be helpful https://github.com/adobe/aepsdk-core-ios/blob/main/AEPIntegrationTests/LifecycleIntegrationTests.swift#L45